src/main/java/com/wanderersoftherift/wotr/init/WotrAttachments.java

```diff
@@ -83,6 +87,12 @@ public class WotrAttachments {
             "ability_states",
             () -> AttachmentType.builder(AbilityStates::new).serialize(AbilityStates.getSerializer()).build());
 
+    public static final Supplier<AttachmentType<? extends AbilityTracker>> ABILITY_TRACKER = ATTACHMENT_TYPES.register(
+            "ability_tracker",
+            () -> AttachmentType.builder(AbilityTracker::new)
+                    .serialize(AbilityTracker.getSerializer())
```

Comment:
I think you could probably just drop the serialize and copy on death, given there is nothing to serialize and nothing to copy.